### PR TITLE
rauc-hawkbit-updater: refactoring

### PR DESCRIPTION
**Pull request by Bastian-Krause**
This is the next PR of a series of PRs to refactor and clean up the code base, get rid of bugs (memory leaks, use-after-frees, race conditions), simplify functions and document them.

In this iteration I focused on the `rauc-hawkbit-updater` module.

### CODE REVIEW

**Review by SijmenHuizenga — APPROVED** [review]

File: src/rauc-hawkbit-updater.c [hunk 1]
```diff
@@ -62,26 +62,30 @@ static GSourceFunc notify_hawkbit_install_complete;
 
 
 /**
- * @brief RAUC callback when new progress.
+ * @brief Callback for install thread, consumes RAUC progress messages, logs them and passes them
+ * on to notify_hawkbit_install_progress().
+ *
+ * @param[in] data install_context pointer allowing access to received status messages
+ * @return G_SOURCE_REMOVE is always returned
```
Comment (ejoerns) [inline src/rauc-hawkbit-updater.c:69]: Semantically, this is a `GSourceFunc()`, maybe this is worth being added to the documentation?

**Review by ejoerns — DISMISSED** [review]
Looks fine for me :+1:

File: src/rauc-hawkbit-updater.c (hunk 1 above)
Comment (Bastian-Krause) [reply src/rauc-hawkbit-updater.c:69]: Yes, I've added this info to all `GSourceFunc`s now.

**Comment by ejoerns** [comment]
Note: Failed CI testing requires the fix in #74 to be merged.

**Comment by Bastian-Krause** [comment]
Rebased on current master.

**Review by ejoerns — APPROVED** [review]

**Pull request merged**